src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletService.scala

```diff
@@ -405,16 +407,21 @@ class ErgoWalletServiceImpl(override val ergoSettings: ErgoSettings) extends Erg
     boxes.map(tb => WalletBox(tb, currentHeight)).sortBy(_.trackedBox.inclusionHeightOpt)
   }
 
-  override def getScanUnspentBoxes(state: ErgoWalletState, scanId: ScanId, considerUnconfirmed: Boolean): Seq[WalletBox] = {
+  override def getScanUnspentBoxes(state: ErgoWalletState, scanId: ScanId, considerUnconfirmed: Boolean, minHeight: Int, maxHeight: Int): Seq[WalletBox] = {
     val unconfirmed = if (considerUnconfirmed) {
-      state.offChainRegistry.offChainBoxes.filter(_.scans.contains(scanId))
+      state.offChainRegistry.offChainBoxes
+        .filter { box =>
+          box.scans.contains(scanId) &&
+          box.inclusionHeightOpt.getOrElse(0) >= minHeight  &&
```

Comment:
for unconfirmed boxes, inclusion height is always defined

I think you need to return unconfirmed with old filter, and now only if maxHeight is not defined by a user

Reply:
I don't understand this comment, old filter was `_.scans.contains(scanId)` and now we also want to filter out boxes unconfirmed boxes that are out of height range :thinking:

Reply:
Unconfirmed boxes do not have inclusion height, they are not in the chain yet

Reply:
I see, we could think about it also in a way that `considerUnconfirmed` overrules the height range ...

Reply:
I think we need to allow considerUnconfirmed to be set only if there's no limit for maxHeight.